[import-w3c-tests] references should be imported as their original filename too

gsnedders opened the pull request:
#### ec1b41b9a866de879a036f55d96bc75bb9b548df
<pre>
[import-w3c-tests] references should be imported as their original filename too
<a href="https://bugs.webkit.org/show_bug.cgi?id=273235">https://bugs.webkit.org/show_bug.cgi?id=273235</a>

Reviewed by Tim Nguyen.

Prior to 277673@main we did import _some_ references under their
original name, as well as copying them to -expected.html.

In that commit, we inadvertently stopped the copying; it was somewhat
surprising behaviour of the existing code that led to some of them being
copied, and arguably a bug, and the obvious rewrite of it led to them
being excluded again.

However, preserving both the original and r-w-t style filenames provides
a clearer path to running the tests based on their &lt;link rel=match&gt;,
thus there is a good reason to do this. (And with git being based on
content addressable storage, there is little cost in duplicating the
files.)

This removes the branch that was always intended to exclude them, and
adds tests to verify we have the now-intended behaviour.

With that now done, the only usage of the `referencefile` metadata is
to exclude reftests which point at themself as the reference, which is
behaviour upstream WPT doesn&apos;t prohibit (but is clearly meaningless!),
so we may as well get rid of the `referencefile` metadata entirely and
just treat references identically to support files.

* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.find_importable_tests):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
* Tools/Scripts/webkitpy/w3c/test_parser.py:
(TestParser.analyze_test):
* Tools/Scripts/webkitpy/w3c/test_parser_unittest.py:
(test_reference_self_reference):
(test_reference_name):

Canonical link: <a href="https://commits.webkit.org/278271@main">https://commits.webkit.org/278271@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1acefd1d0539f11f6cf89668040e44eaa588fabd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitpy ](https://ews-build.webkit.org/#/builders/WebKitPy-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 gtk-wk2 ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 services ](https://ews-build.webkit.org/#/builders/Services-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-10-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7222 "Built successfully and passed tests") | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-10-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Tools/Scripts/webkitpy/w3c/test_parser.py:
```diff
@@ -122,8 +122,6 @@ def analyze_test(self, test_contents=None, ref_contents=None):
             test_info = {'test': self.filename, 'crashtest': True}
         elif test_type == "testharness":
             test_info = {'test': self.filename, 'jstest': True}
-        elif test_type == "support" and wpt_sourcefile.name_is_reference:
-            test_info = {'referencefile': self.filename}
```
Isn't this metadata useful to have around? Would it be better to delete this instead: https://searchfox.org/wubkat/rev/088a3e573b59664dc988291e52900bba6b106b2c/Tools/Scripts/webkitpy/w3c/test_importer.py#387-389 ?

gsnedders replied:
Why would we need the metadata? It's… meaningless. Though it does make me realise that maybe we should just totally remove `referencefile`; there's nothing meaningfully different with them versus arbitrary support files.

gsnedders replied:
OK, now actually totally removed the metadata (and updated the commit message to reflect that).

gsnedders committed:
[import-w3c-tests] references should be imported as their original filename too
https://bugs.webkit.org/show_bug.cgi?id=273235

Reviewed by Tim Nguyen.

Prior to 277673@main we did import _some_ references under their
original name, as well as copying them to -expected.html.

In that commit, we inadvertently stopped the copying; it was somewhat
surprising behaviour of the existing code that led to some of them being
copied, and arguably a bug, and the obvious rewrite of it led to them
being excluded again.

However, preserving both the original and r-w-t style filenames provides
a clearer path to running the tests based on their <link rel=match>,
thus there is a good reason to do this. (And with git being based on
content addressable storage, there is little cost in duplicating the
files.)

This removes the branch that was always intended to exclude them, and
adds tests to verify we have the now-intended behaviour.

With that now done, the only usage of the `referencefile` metadata is
to exclude reftests which point at themself as the reference, which is
behaviour upstream WPT doesn't prohibit (but is clearly meaningless!),
so we may as well get rid of the `referencefile` metadata entirely and
just treat references identically to support files.

* Tools/Scripts/webkitpy/w3c/test_importer.py:
(TestImporter.find_importable_tests):
* Tools/Scripts/webkitpy/w3c/test_importer_unittest.py:
* Tools/Scripts/webkitpy/w3c/test_parser.py:
(TestParser.analyze_test):
* Tools/Scripts/webkitpy/w3c/test_parser_unittest.py:
(test_reference_self_reference):
(test_reference_name):

Canonical link: https://commits.webkit.org/278271@main
